src/Agent.Listener/Configuration/WindowsRegistryHelper.cs

```diff
@@ -0,0 +1,206 @@
+#if OS_WINDOWS
+using System;
+using System.Security.Principal;
+using Microsoft.Win32;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(WindowsRegistryManager))]
+    public interface IWindowsRegistryManager : IAgentService
+    {
+        string GetKeyValue(string path, string keyName);
+        void SetKeyValue(string path, string keyName, string keyValue);
+        void DeleteKey(string path, string keyName);
+
+        bool RegsitryExists(string securityId);
+    }
+
+    public class WindowsRegistryManager : AgentService, IWindowsRegistryManager
+    {
+        public void DeleteKey(string path, string keyName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public string GetKeyValue(string path, string keyName)
+        {
+            var regValue = Registry.GetValue(path, keyName, null);
+            return regValue != null ? regValue.ToString() : null;
+        }
+
+        public void SetKeyValue(string path, string keyName, string keyValue)
+        {
+            Registry.SetValue(path, keyName, keyValue, RegistryValueKind.String);
+        }
+
+        public bool RegsitryExists(string securityId)
+        {
+            return Registry.Users.OpenSubKey(securityId) != null;
+        }
+    }
+
+    public class WindowsRegistryHelper
```

Comment:
RegistryManager just deals with paths, keys and the values. Registry helper is one layer above which is transforming the well known registries into the actual paths and keys. Path may vary as per user profile/OS profile. Registry Helper is also going to help in taking the backup of the registry so that UnConfigure() can revert back the original values. Shall I move this Helper into InteractiveConfigSessionManager class?